inputremapper/injection/context.py

```diff
@@ -84,6 +110,14 @@ def update_purposes(self):
         self.left_purpose = self.mapping.get("gamepad.joystick.left_purpose")
         self.right_purpose = self.mapping.get("gamepad.joystick.right_purpose")
 
+    def create_callbacks(self) -> None:
+        """add the notify method from all _handlers to self.callbacks"""
+        for key, handler_list in self._handlers.items():
+            if key[0][:2] not in self.callbacks.keys():
+                self.callbacks[key[0][:2]] = []
+            for handler in handler_list:
+                self.callbacks[key[0][:2]].append(handler.notify)
```

Comment:
could this also use `.listeners`?

how would it be possible to ensure the listeners for macros run first? Via a priority parameter?